docs/ignoring-files-folders-code.md

```diff
@@ -0,0 +1,186 @@
+---
+slug: ignoring-files-folders-code 
+append_help_link: true
+title: Ignoring files, folders, or code
+description: "This documents various methods to skip or ignore files, folders, or code that are not relevant to a Semgrep scan."
+---
+
+import MoreHelp from "/src/components/MoreHelp"
+
+
+# Ignoring files, folders, or code
+
+**Ignoring files, folders, or code** means Semgrep will not scan certain defined items within your repository or working directory. Thus no findings are generated.
+
+This is separate from [ignoring a **finding**](../managing-findings), which is a triage operation after a Semgrep scan has found a match.
+
+All Semgrep environments (CLI, CI, and App) adhere to user-defined or Semgrep-defined ignore patterns.
+
+
+## Reference summary
+
+| Method  | Usage    | Examples |
+|:--------|:---------|:---------|
+| To ignore blocks of code: `nosemgrep` | Create a comment, followed by a space (` `), followed by `nosemgrep` at the first line or preceding line of the pattern match. | ` // nosemgrep` &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; `// nosemgrep: rule-id` <br /> `# nosemgrep` |
+| To ignore files and folders: `.semgrepignore` | Create a `.semgrepignore` file and add patterns for files and folders there. Patterns follow `.gitignore` syntax with some caveats. See [Defining files and folders in `.semgrepignore`](#defining-files-and-folders-in-semgrepignore). | [`.semgrepignore` sample file](https://raw.githubusercontent.com/returntocorp/semgrep/develop/semgrep/semgrep/templates/.semgrepignore) |
+
+
+## Understanding Semgrep defaults
+
+Without user customization, Semgrep refers to the following to define ignored files and folders:
+
+* Semgrep's default `.semgrepignore` file
```

Comment:
One thing that is not clear to me about the .semgrepignore file is where it comes from and where it should be placed. As written here, I could understand that Semgrep (the program) has a .semgrepignore file and as a user I get it when I install Semgrep. But I think that’s not the case? I think this default file refers to the template we have in the Semgrep, repo, right?

If I understand correctly, then the user does need to take some action (perhaps not customization, strictly speaking) to make use of this default file.

Reply:
I've replaced
> Semgrep's default `.semgrepignore` file:

with
>In the absence of a user-generated `.semgrepignore`, Semgrep will refer to its repository's `.semgrepignore` default template:

To make it explicit "where" this default is coming from. Please let me know if this makes sense?

Reply:
Yes, thank you!